### PR TITLE
Remove last_plan_solution_ in PlanningComponent

### PR DESCRIPTION
**Pull request by sjahr**
### Description

Addresses discussion in #1862 

BREAKING CHANGE

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/planning/moveit_cpp/src/planning_component.cpp [hunk 1]
```diff
@@ -398,25 +376,7 @@ bool PlanningComponent::setGoal(const std::string& goal_state_name)
...
-  //  RCLCPP_ERROR("Failed to parameterize trajectory");
-  //  return false;
-  //}
-  return moveit_cpp_->execute(last_plan_solution_.trajectory_, blocking);
-}
-
-const planning_interface::MotionPlanResponse& PlanningComponent::getLastMotionPlanResponse()
-{
-  return last_plan_solution_;
+  RCLCPP_ERROR(LOGGER, "There is no successful plan to execute");
```
Comment (sjahr) [inline moveit_ros/planning/moveit_cpp/src/planning_component.cpp:379]: I am wondering whether or not this means we can remove the whole function :thinking:

File: moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/planning_component.h [hunk 2]
```diff
@@ -236,12 +232,3 @@ class PlanningComponent
   // std::unique_ptr<moveit_msgs::msg::TrajectoryConstraints> trajectory_constraints_;
 };
 }  // namespace moveit_cpp
-
-namespace moveit
```
Comment (sjahr) [inline moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/planning_component.h:240]: This is deprecated for 22 months. I'd remove it with this PR but I can also revert this change if you don't feel so

File: moveit_ros/planning/moveit_cpp/src/planning_component.cpp (hunk 1 above)
Comment (AndyZe) [reply moveit_ros/planning/moveit_cpp/src/planning_component.cpp:379]: I would say so

**Review by AndyZe — COMMENTED** [review]
Looks good to me besides removing that last function

File: moveit_ros/planning/moveit_cpp/src/planning_component.cpp (hunk 1 above)
Comment (AndyZe) [reply moveit_ros/planning/moveit_cpp/src/planning_component.cpp:379]: Prob should add a deprecation warning

File: moveit_ros/planning/moveit_cpp/src/planning_component.cpp [hunk 3]
```diff
@@ -396,27 +374,9 @@ bool PlanningComponent::setGoal(const std::string& goal_state_name)
...
-  // max_acceleration_scaling_factor_))
-  //{
-  //  RCLCPP_ERROR("Failed to parameterize trajectory");
-  //  return false;
-  //}
-  return moveit_cpp_->execute(last_plan_solution_.trajectory_, blocking);
-}
-
-const planning_interface::MotionPlanResponse& PlanningComponent::getLastMotionPlanResponse()
+bool PlanningComponent::execute(bool /*blocking*/)
```
Comment (henningkayser) [inline moveit_ros/planning/moveit_cpp/src/planning_component.cpp:377]: we should really drop the blocking flag altogether.

**Review by henningkayser — APPROVED** [review]

File: moveit_ros/planning/moveit_cpp/src/planning_component.cpp (hunk 3 above)
Comment (AndyZe) [reply moveit_ros/planning/moveit_cpp/src/planning_component.cpp:377]: The "simultaneous execution" PR to ROS2 eventually (I hope so!). It looks like it does use a blocking flag in MoveItCpp but not the blocking flag in PlanningComponent. So I agree, the `blocking` flag here could probably be dropped.

https://github.com/ros-planning/moveit/pull/3243/files

**Comment by sjahr** [comment]
@AndyZe I removed the execute function from the source file but kept it barebone in the header so the deprecation warning will still be there. Does this work for you or is there an even smarter way?

File: moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/planning_component.h [hunk 4]
```diff
@@ -202,10 +202,10 @@ class PlanningComponent
 
   /** \brief Execute the latest computed solution trajectory computed by plan(). By default this function terminates
    * after the execution is complete. The execution can be run in background by setting blocking to false. */
-  [[deprecated("Use MoveItCpp::execute()")]] bool execute(bool blocking = true);
-
-  /** \brief Return the last plan solution*/
-  const planning_interface::MotionPlanResponse& getLastMotionPlanResponse();
```
Comment (AndyZe) [inline moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/planning_component.h:208]: @sjahr I think you might have screwed up the previous commit. Didn't you mean to add a deprecation warning here?

Comment (sjahr) [reply moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/planning_component.h:208]: I'll fix that

Comment (sjahr) [reply moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/planning_component.h:208]: :thinking: After giving it a second thought, I am unsure how I can deprecate this function in a good way. It returns a reference to a not existing member variable. If I'd keep the member variable the whole point of the PR is gone. @AndyZe Should we keep it this way? It is either way a breaking change

Comment (AndyZe) [reply moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/planning_component.h:208]: I see. Well, let's just delete it then. The nice thing about Rolling is we don't have to worry too much about API breakage  :+1:

**Review by AndyZe — APPROVED** [review]

**Pull request merged**